analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -37,6 +39,11 @@ public sealed record SymbolicValue
         public IEnumerable<SymbolicConstraint> AllConstraints =>
             Constraints.Values;
 
+        public SymbolicValue()
+        {
+            hashCode = new(() => HashCode.DictionaryContentHash(Constraints), LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
For `LazyThreadSafetyMode` see https://learn.microsoft.com/en-us/dotnet/api/system.threading.lazythreadsafetymode?view=net-8.0